### ISSUE TITLE
Consider using -strict-sequence

### ISSUE THREAD

**Issue by hhugo** [ocaml-community/yojson#165]
or even rely on dune default flags (plus adjustments).

dune default's are currently completely ignored because of
```sexp
(flags (-safe-string)))
```

I would suggest to use the following instead
```
(flags (:standard -w -27-32-34))
```
- `safe-string` is the default nowdays. `unsafe-string` was completely removed starting with ocaml 5.0.  
- warnings 27, 32, 34 currently trigger but are harder to fix in the current workflow. 

Dune `:standard` include `-strict-sequence` witch trigger the following error. It's unclear to me which of the implementation or the signature should be fixed. 

```
$ (cd _build/default && /home/hugo/tezos/_opam/bin/ocamlopt.opt -strict-sequence -safe-string -g -I lib/.yojson.objs/byte -I lib/.yojson.objs/native -I /home/hugo/tezos/_opam/lib/seq -intf-suffix .ml -no-alias-deps -opaque -o lib/.yojson.objs/native/yojson.cmx -c -impl lib/yojson.ml)
File "write.ml", line 428, characters 2-62:
Error: This expression has type int but an expression was expected of type
         unit
       because it is in the left-hand side of a sequence
Command [2] exited with code 2:              
$ (cd _build/default && /home/hugo/tezos/_opam/bin/ocamlc.opt -strict-sequence -safe-string -g -bin-annot -I lib/.yojson.objs/byte -I /home/hugo/tezos/_opam/lib/seq -intf-suffix .ml -no-alias-deps -opaque -o lib/.yojson.objs/byte/yojson.cmo -c -impl lib/yojson.ml)
File "write.ml", line 428, characters 2-62:
Error: This expression has type int but an expression was expected of type
         unit
       because it is in the left-hand side of a sequence
```

**Comment by Leonidas-from-XIV** [comment]
Hi @hhugo, you're raising some good points. We should definitely have the default flags enabled - I think the reason they are set like this is mostly that they're missing `:standard` by accident.

However upon thinking about it I realized we use `safe-string` is because we still support 4.02 and `safe-string` only became the default in 4.06, so it can't really be dropped yet I think.

I looked at `to_output` and you're right. I think the interface is wrong as the code never does anything with the `int` except discard it. `safe-sequence` turns this into an error. However changing the interface is a breaking change, so despite probably nobody using it, I'd rather not change this without a major version bump (for which I'd like to queue up a few more changes).

I've created #166 to improve the current situation.
